pyproject.toml

```diff
@@ -104,6 +104,10 @@ disable_error_code = ["arg-type", "operator", "union-attr"]
 module = "data_store/validation/towns_fund/fund_specific_validation/fs_validate_r4"
 disable_error_code = ["arg-type", "assignment"]
 
+[[tool.mypy.overrides]]
+module = "data_store/validation/towns_fund/fund_specific_validation/fs_validate_r6"
+disable_error_code = ["arg-type"]
```

Comment:
Please don't disable the error.

Checking the error. You have one that needs fixing:
```
data_store/validation/towns_fund/fund_specific_validation/fs_validate_r6.py:55: error: Argument "row_index" to "GenericFailure" has incompatible type "Hashable | None"; expected "int | None"  [arg-type]
```

Only change needed is:
`                    row_index=idx,`
to
`                    row_index=tpying.cast(int, idx),` (`import typing` at the top)

row_index is very likely an int. `typing.cast` makes python treat it as a int.

Reply:
makes mypy consider it an int - doesn't change anything about how python treats it in code

Reply:
I think we should have a convo at the next tech chat around where that 'useful type hints' boundary lies. We should definitely be prioritising type hints where they add value/safety, not aiming for 100% coverage for the sake of it (especially around pandas where the type hints will get really gnarly).

Reply:
- `idx` is of type `Hashable | None`
- `GenericFailure` is accepting a `row_index` of type `int | None`
One of the two sides would need to be changed if we want to fix it properly.

`typing.cast` is indeed just treating the symptom. 
If you wish to ignore it, I would still rather suggest you inline ignore it. (`  # type: ignore[arg-type]`)

Reply:
I changed my mind. I thought I could rid the code base of mypy exceptions. But at what cost?!

The file exception is fine, I left a few exceptions myself.

Reply:
Inline ignoring specific things that interact with pandas feels yeah OK (defo better to inline single exceptions rather than file ignore, unless there are loads)